### DISCUSSION TITLE
Login landing page

### DISCUSSION THREAD

**Discussion by mikeprag** [oqtane/oqtane.framework#881] [Q&A]
Hi All,

Just starting out using Oqtane and wondered if someone could help.

I want to have all pages needing authentication so when the user lands on the site, it always goes to the login page.

I also want this page to look different from the rest of the site (split screen with image on left and login box on right).

Could someone give me some direction on how to do this or point me at some resource to follow (if it exists).

Really appreciate any help

Mike

**Comment by hishamco** [comment]
> I want to have all pages needing authentication so when the user lands on the site

Simply you can set the role for all the page to "Authenticated Users"

> I also want this page to look different from the rest of the site (split screen with image on left and login box on right).

This required you to modify the theme

**Reply by mikeprag** [reply]
Thanks for the quick response!

Is there some resource for developing or altering a theme?

Thanks

Mike

**Reply by hishamco** [reply]
@sbwalker @ADefWebserver did anyone of you blogged about creating a custom theme?

**Reply by sbwalker** [reply]
Use the Control Panel icon in the upper right corner, choose Page Management - Edit, select the Permissions tab, uncheck the box for View permissions for All Users and Save. This will instruct the framework that a user must be logged in to view it. You will need to do this for all page EXCEPT for the Login page. You must also set the Permissions in this manner when creating new pages.

You will need a theme with 2 panes. Are you planning on using the default Oqtane theme (black/blue) or a different theme?

**Reply by sbwalker** [reply]
I dont think anyone has created a blog yet about how to create a custom theme. There is a custom theme project you can look at for reference https://github.com/oqtane/oqtane.theme.cerulean.

**Reply by hishamco** [reply]
I think we should focus on docs little bit in upcoming days to give a good start in some of the Oqtane fundemantels

**Reply by mikeprag** [reply]
Probably going to use a different theme I think - It's also not my thing but I can probably just about get by!

I'm creating a portal for users so would find it very useful to have a tutorial or best practices on how to create a page with different widgets that you can add as modules to the page..Something like the dashboard page of SBAdmin site

https://startbootstrap.com/previews/sb-admin-2

I'm also planning to use some of the Blazor components from Syncfusion so what would be the best way to include 3rd party components.

Finally, MFA would be very useful.

Apologies if these have already been converted elsewhere - Just trying to find as much about Oqtane as possible.

Thanks for all the replies and help

Mike

**Reply by mikeprag** [reply]
Hi Sean,

Few questions:

1) I downloaded the theme project. I kind of see what's going on but to create the project - Did you go through the module creation or just create a new project in VS or some other way?
2) In the main Oqtane.Server / wwwroot / Themes / Oqtane.Themes.OqtaneTheme folder, there is a BootswatchCyborg.css file but in the theme project, this isn't onethere - Am I missing something?
3) From the bootswatch themes, is BootswatchCyborg.css just the bootstrap.css from the theme?

Might need a little overall guidance on creating a theme from scratch

Thanks in advance

Mike

**Comment by ADefWebserver** [comment]
@hishamco - Theming is something I have no talent in :) So I won’t be doing that one...

**Reply by hishamco** [reply]
No problem I can do this instead ;)
